### PR TITLE
Add printed_sales_invoice and layout resources

### PR DESCRIPTION
**Pull request by commuun**

### CODE REVIEW

File: lib/elmas/resources/printed_sales_invoice.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  # Open to 'Verwerkt' while at the same time sending a PDF of the invoice to the
+  # end user by e-mail.
+  #
+  # This endpoint only supports the POST method.
+  #
+  class PrintedSalesInvoice
+    include Elmas::Resource
+
+    def valid_actions
+      [:post]
```
Comment (Marthyn) [inline lib/elmas/resources/printed_sales_invoice.rb]: Only post and no get? @commuun

File: spec/resources/invoices/printed_sales_invoice_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  context "Applying filters" do
+    resource = Elmas::PrintedSalesInvoice.new(id: 123)
+    it "should apply ID filter for find" do
+      expect(Elmas).to receive(:get).with("salesinvoice/PrintedSalesInvoices(guid'123')?")
+      resource.find
+    end
+
+    it "should apply no filters for find_all" do
+      resource = Elmas::PrintedSalesInvoice.new(type: 2)
+      expect(Elmas).to receive(:get).with("salesinvoice/PrintedSalesInvoices?")
```
Comment (Marthyn) [inline spec/resources/invoices/printed_sales_invoice_spec.rb]: also this spec should fail then if only post is allowed *my fault*

File: lib/elmas/resources/printed_sales_invoice.rb (hunk 1 above)
Comment (commuun) [inline lib/elmas/resources/printed_sales_invoice.rb]: That's right, the name can be a bit confusing. This is not for retrieving a document, it's for the one-time generation of the invoice PDF (and optionally sending it to the end user by e-mail) It changes the invoice's state, and can only be called once for each invoice.

Comment (Marthyn) [inline lib/elmas/resources/printed_sales_invoice.rb]: Allright.

**Comment by Marthyn** [comment]
I'll merge this but i need to look at specs that should actually fail, when only `:post` is allowed the find_all spec should def fail.

**Pull request merged**